ff/handlers/nonbonded.py

```diff
@@ -161,6 +162,82 @@ def parameterize_ligand(params, param_idxs):
     return params[param_idxs]
 
 
+def compute_or_load_am1_charges(mol):
+    """Unless already cached in mol's "AM1Cache" property, use OpenEye to compute AM1 partial charges."""
+
+    # check for cache
+    cache_key = "AM1Cache"
+    if not mol.HasProp(cache_key):
+        # The charges returned by OEQuacPac is not deterministic across OS platforms. It is known
+        # to be an issue that the atom ordering modifies the return values as well. A follow up
+        # with OpenEye is in order
+        # https://github.com/openforcefield/openff-toolkit/issues/983
+        am1_charges = list(oe_assign_charges(mol, "AM1"))
+
+        mol.SetProp(cache_key, base64.b64encode(pickle.dumps(am1_charges)))
+
+    else:
+        am1_charges = pickle.loads(base64.b64decode(mol.GetProp(cache_key)))
+
+    return np.array(am1_charges)
+
+
+def bond_smirks_matches(mol, smirks_list):
+    """Return an array of ordered bonds and an array of their assigned types
+
+    Notes
+    -----
+    * Uses OpenEye for substructure searches
+    * "First match wins" -- e.g. if bond (a,b) can be matched by smirks_list[2], smirks_list[5], ..., only return 2
+    """
+    oemol = convert_to_oe(mol)
+    AromaticityModel.assign(oemol)
+
+    bond_idxs = []  # [B, 2]
+    type_idxs = []  # [B]
+
+    for type_idx, smirks in enumerate(smirks_list):
+        matched_bonds = []
+
+        matches = oe_match_smirks(smirks, oemol)
+
+        for matched_indices in matches:
+            a, b = matched_indices[0], matched_indices[1]
+            forward_matched_bond = [a, b]
+            reverse_matched_bond = [b, a]
```

Comment:
Does forward/reverse order of bonds matter in the result? If not, could this be simplified by putting the lesser index first?
```suggestion
            matched_bond = sorted([a, b])
```

Reply:
The order does matter here, I think!

E.g. for a pair of atoms that is assigned this type

https://github.com/proteneer/timemachine/blob/d7acd339a681618fd2819b072f48f5aaafcf3b1b/ff/charges.py#L3

the partial charge of the carbon ("#6X4") will be adjusted in the [positive](https://github.com/proteneer/timemachine/blob/d1ed0e67ebe7a4b898b5104c3ab18bcfa81bd929/ff/handlers/nonbonded.py#L233) direction, but the partial charge of the sulfur "#16X1-1" will be adjusted in the [negative](https://github.com/proteneer/timemachine/blob/d1ed0e67ebe7a4b898b5104c3ab18bcfa81bd929/ff/handlers/nonbonded.py#L234) direction. Will add a note to clarify

Reply:
Added notes about this in two places: https://github.com/proteneer/timemachine/pull/571/commits/ff2c6b154e50991952cdc10911daf912a9fa30c4

Reply:
Makes sense, thanks for clarifying!